lib/cookie.js

```diff
@@ -869,15 +901,12 @@ CookieJar.prototype.setCookie = function(cookie, url, options, cb) {
     cookie.domain = host;
   }
 
-  // S5.3 step 7: "Otherwise, set the cookie's path to the default-path of the
-  // request-uri"
-  if (!cookie.path) {
+  //S5.2.4 If the attribute-value is empty or if the first character of the
+  //attribute-value is not %x2F ("/"):
+  //Let cookie-path be the default-path.
+  if (!cookie.path || cookie.path[0] !== '/') {
```

Comment:
S5.2.4 cases are: `Path=;` and `Path=anything_except_slash`. S5.3.7 covers case when we don't have path attribute at all. In all those cases we should fallback to the default-path. So, yes this code will cover both S5.2.4 and S5.3.7.